state/indexer/doc.go

```diff
@@ -0,0 +1,74 @@
+/*
+Package indexer defines Tendermint's block and transaction event indexing logic.
+
+Currently, Tendermint supports two primary means of block and transaction event
+indexing:
+
+1. A key-value sink via an embedded database with a proprietary query language.
+2. A Postgres-based sink.
+
+Recall, an ABCI application can emit events during block and transaction execution
+in the form of <abci.Event.Type>.<abci.EventAttributeKey>=<abci.EventAttributeValue>,
+e.g. 'transfer.amount=10000'.
+
+An operator can enable one or both of the supported indexing sinks via the
+'tx-index.indexer' Tendermint configuration.
+
+Example:
+
+	[tx-index]
+	indexer = ["kv", "psql"]
+
+If an operator wants to completely disable indexing, they may simply just provide
+the "null" sink option in the configuration. All other sinks will be ignored if
+"null" is provided.
+
+If indexing is enabled, the indexer.Service will iterate over all enabled sinks
+and invoke block and transaction indexing via the appropriate IndexBlockEvents
+and IndexTxEvents methods.
+
+Note, the "kv" sink is considered deprecated and it's query functionality is very
+limited, but does allow users to directly query for block and transaction events
+against Tendermint's RPC. Instead, operators are encouraged to use the "psql"
+indexing sink when more complex queries are required and for reliability purposes
+as PostgreSQL can scale.
+
+Prior to starting Tendermint with the "psql" indexing sink enabled, operators
+must ensure the following:
+
+1. The "psql" indexing sink is provided in Tendermint's configuration.
+2. A 'tx-index.psql-conn' value is provided that contains the PostgreSQL connection URI.
```

Comment:
I'm not sure if there's a rule behind when you're using 'single quotes' vs. "double quotes" but I'd suggest sticking to just one of them (personally I like double, but I'm not picky).